src/renderer/components/dialog-add-version.tsx

```diff
@@ -168,25 +190,33 @@ export class AddVersionDialog extends React.Component<
   }
 
   private renderPath(): JSX.Element | null {
-    const { folderPath, isValidElectron } = this.state;
+    const { isValidElectron, folderPath, existingLocalVersion } = this.state;
+    const shouldSetToPreviousVersion = isValidElectron && existingLocalVersion;
 
     if (!folderPath) return null;
-
-    const info = isValidElectron
-      ? `We found an ${getElectronNameForPlatform()} in this folder.`
-      : `We did not find a ${getElectronNameForPlatform()} in this folder...`;
-
     return (
       <Callout>
-        {info}
-        {this.renderVersionInput()}
+        {this.buildDialogText()}
+        {!shouldSetToPreviousVersion && this.renderVersionInput()}
       </Callout>
     );
   }
 
+  private buildDialogText(): string {
+    const { existingLocalVersion, isValidElectron } = this.state;
+
+    if (isValidElectron && existingLocalVersion) {
```

Comment:
Since we've got the same of (isValidElectron && existingLocalVesion) logic repeated here again, I'd probably use the same variables here too for clarity's sake, e.g.

```
const canAdd = isValidElectron && isValidVersion && !existingLocalVersion;
const canSwitch = isValidElectron && existingLocalVersion;

if (canSwitch) {
  return ...
}
if (canAdd) {
  return ...
}
return ...
```

Reply:
I apologize if I'm mistaken, but I took a look at this with @VerteDinde for a bit on Friday and to my understanding, we were unable to make variable names work like you described here due to some typescript issues like so: 

![image](https://user-images.githubusercontent.com/33054982/123678170-d6c59180-d7fa-11eb-94e1-00bef999a9fa.png)

Reply:
@georgexu99 this is more a nice-to-have and not a blocker.

However, in that screenshot I think at line 211 we know that `existingLocalVersion` is defined because of the test on line 209. So we just have to explain that to TypeScript by appending a `!` to the variable name e.g.

```diff
- ${existingLocalVersion.version}
+ ${existingLocalVersion!.version}
```

Reply:
oh Thank you!